crates/sui-core/src/consensus_adapter.rs

```diff
@@ -170,18 +180,127 @@ impl SubmitToConsensus for TransactionsClient<sui_network::tonic::transport::Cha
     }
 }
 
+pub struct ConnectionMonitorStatusForTests {}
+
+impl CheckConnection for ConnectionMonitorStatusForTests {
+    fn check_connection(&self, _authority: &AuthorityName) -> Option<ConnectionStatus> {
+        Some(ConnectionStatus::Connected)
+    }
+}
+
+pub trait CheckConnection {
+    fn check_connection(&self, authority: &AuthorityName) -> Option<ConnectionStatus>;
+}
+
+pub struct ConnectionMonitorStatus {
+    /// Current connection statuses forwarded from the connection monitor
+    pub connection_statuses: Arc<DashMap<AuthorityName, ConnectionStatus>>,
+    /// Join handle for the Connection Monitor Listener
+    pub connection_monitor_listener_handle: JoinHandle<()>,
+}
+
+pub struct ConnectionMonitorListener {
```

Comment:
Could you please add some test coverage for this?

Reply:
I've now removed this